client/cmd/dexc-webview/config.go

```diff
@@ -0,0 +1,263 @@
+	TorIsolation bool   `long:"torisolation" description:"Enable TOR circuit isolation."`
+	Net          dex.Network
+	CertHosts    []string
+}
+
+var defaultConfig = Config{
+	AppData:    defaultApplicationDirectory,
+	Config:     defaultConfigPath,
+	DebugLevel: defaultLogLevel,
+	CertHosts: []string{defaultTestnetHost, defaultSimnetHost,
+		defaultMainnetHost},
+}
+
+// configure processes the application configuration.
+func configure() (*Config, error) {
+	// Pre-parse the command line options to see if an alternative config file
+	// or the version flag was specified. Override any environment variables
+	// with parsed command line flags.
+	iniCfg := defaultConfig
+	preCfg := iniCfg
+	preParser := flags.NewParser(&preCfg, flags.HelpFlag|flags.PassDoubleDash)
+	_, flagerr := preParser.Parse()
+
+	if flagerr != nil {
+		e, ok := flagerr.(*flags.Error)
+		if !ok || e.Type != flags.ErrHelp {
+			preParser.WriteHelp(os.Stderr)
+		}
+		if ok && e.Type == flags.ErrHelp {
+			preParser.WriteHelp(os.Stdout)
+			os.Exit(0)
+		}
+		return nil, flagerr
+	}
+
+	// Show the version and exit if the version flag was specified.
+	if preCfg.ShowVer {
+		fmt.Printf("%s version %s (Go version %s %s/%s)\n",
+			version.AppName, version.Version(), runtime.Version(), runtime.GOOS, runtime.GOARCH)
+		os.Exit(0)
+	}
+
+	// If the app directory has been changed, but the config file path hasn't,
+	// reform the config file path with the new directory.
+	if preCfg.AppData != defaultApplicationDirectory && preCfg.Config == defaultConfigPath {
+		preCfg.Config = filepath.Join(preCfg.AppData, configFilename)
+	}
+	cfgPath := cleanAndExpandPath(preCfg.Config)
+
+	// Load additional config from file.
+	parser := flags.NewParser(&iniCfg, flags.Default)
+	err := flags.NewIniParser(parser).ParseFile(cfgPath)
+	if err != nil {
+		if _, ok := err.(*os.PathError); !ok {
+			fmt.Fprintln(os.Stderr, err)
+			parser.WriteHelp(os.Stderr)
+			return nil, err
+		}
+		// Missing file is not an error.
+	}
+
+	// Parse command line options again to ensure they take precedence.
+	_, err = parser.Parse()
+	if err != nil {
+		if e, ok := err.(*flags.Error); !ok || e.Type != flags.ErrHelp {
+			parser.WriteHelp(os.Stderr)
+		}
+		return nil, err
+	}
+
+	// Set the global *Config.
+	cfg = &iniCfg
+
+	if cfg.Simnet && cfg.Testnet {
+		return nil, fmt.Errorf("simnet and testnet cannot both be specified")
+	}
+
+	var defaultDBPath string
+	switch {
+	case cfg.Testnet:
+		cfg.Net = dex.Testnet
+		defaultDBPath = setNet(preCfg.AppData, "testnet")
+	case cfg.Simnet:
+		cfg.Net = dex.Simnet
+		defaultDBPath = setNet(preCfg.AppData, "simnet")
+	default:
+		cfg.Net = dex.Mainnet
+		defaultDBPath = setNet(preCfg.AppData, "mainnet")
+	}
+	defaultHost := defaultHostByNetwork(cfg.Net)
+
+	// If web or RPC server addresses not set, use network specific
+	// defaults
+	if cfg.WebAddr == "" {
+		cfg.WebAddr = net.JoinHostPort(defaultHost, defaultWebPort)
+	}
```

Comment:
We should just allow this to be auto-generated, i.e. `localhost:0`. The `Server.addr` field will still be set properly (since we're reading it from the `net.Listener`), and there's no chance of conflict.